cEP-0021.md

```diff
@@ -0,0 +1,349 @@
+    `reverse_order`,`strip_dirs()`,`sort_stats()` and display it on
+    *CLI* i.e. <br>
+    `coala -b ... --files ... --profile-commands "reverse_order,sort_stats()"`
+    <br> or `profile_commands = reverse_order,sort_stats()`.
+
+- **Using `--profile-dump`**:
+
+  Bear writer will also have an ability to directly dump the raw
+  profile output either on *current working <br> directory* or to a
+  *specified directory name*. (which can be further used for estimation
+  of profiler stats)
+
+  - Using default settings provided by `coala` i.e
+    `coala -b Bear Mybear --files file --profile-output`<br> or
+    `profile_output = True` (it will dump the  dump the raw profile output
+    in *current working directory*).
+  - By passing the directory name through *CLI* i.e. <br>
+    `coala -b Bear Mybear --files file --profile-output [Dirname]`
+    or using <br> `.coafile` i.e. `profile_dump = [Dirname]`
+
+The profiling will start by profiling the `run()` method of the bears
+The advantage of profiling just the `run()` method is that, bear writer
+doesn't have to spend time on loading the files, collecting the settings,
+etc. as this is all done by coala itself, they just have to Profile
+the `run()` method to create a Better Performant bear.
+
+After invoking the profiler on the run method of a bear it will first save
+all the profile output on a local temp file which is further used by
+`pstats module` to sort the profile results and formats them in a way desired
+by the Bear writer.
+
+# Implementation Details
+
+When user invokes the profiler with either `CLI` or by `coafile`
+then `execute` method of Bear base Class will call the `run_bear_from_section`
+method.
+
+```python
+def run_bear_from_section(self, args, kwargs):
+    ............
+    profile_bears_results, args, kwargs = self.setup_profiler(
+        *args, **kwargs)
+    return profile_bears_results if profile_bears_results else self.run(
+        *args, **kwargs)
+```
+
+Which further calls the `setup_profiler()` method, and `setup_profiler()` method
+first enables an instance of the `Profile` class, and then enable the profiler
+on `run()` method of bears then check whether keyword arguments
+passed to the Bear contain the *profiler key* or not.
+
+```python
+if (profile_output and profile_commands and profile_dump) is None:
+    kwargs.pop('profile_output')
+    kwargs.pop('profile_commands')
+    kwargs.pop('profile_dump')
+    return False, args, kwargs
+```
+
+- If `yes` then it will split all the values passed through CLI.<br>For eg.
+  `--profile-commands "reverse_order,strip_dirs(),sort_stats('ncalls')"`
+  or by `coafile` and return a list of all the comma seperated values.
+  i.e. `['reverse_order','strip_dirs','sort_stats('ncalls')']`
+
+- If `No` then it will return `False` and no profiling will be done
+  and coala will continue its normal execution.
+
+  > In both the case i.e. whether Profiler is getting invoked or not
```

Comment:
"In both cases, i.e., whether the profiler is ..."

also I would remove the quotation characters.